lib/src/nuget/nuget.dart

```diff
@@ -0,0 +1,87 @@
+import 'dart:convert';
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:archive/archive_io.dart';
+import 'package:ffi/ffi.dart';
+import 'package:http/http.dart' as http;
+import 'package:win32/win32.dart';
+
+import '../utils/exception.dart';
+
+/// Utilities for working with nuget.org, Microsoft's package manager.
+class NuGet {
+  static const authority = 'api.nuget.org';
+  static const basePath = '/v3-flatcontainer';
+
+  static final String tempDirectory = _getTemporaryDirectory();
+
+  static Future<List<String>> getVersions(String packageName) async {
+    final uri = Uri.https(
+      authority,
+      '$basePath/${packageName.toLowerCase()}/index.json',
+    );
+    final response = await http.get(uri);
+    final decodedResponse = jsonDecode(response.body);
+    if (decodedResponse['versions'] == null) {
+      throw WinmdException('Failed to get the versions of $packageName.');
+    }
+
+    final versions = decodedResponse['versions'] as List;
+    return versions.cast<String>();
+  }
+
+  static Future<String> getLatestVersion(String packageName,
+      {bool includePreviewVersions = false}) async {
+    final versions = await getVersions(packageName);
+
+    if (includePreviewVersions) {
+      return versions.last;
+    } else {
+      // Otherwise, return the latest stable version.
+      return versions.lastWhere((version) => !version.contains('preview'));
+    }
```

Comment:
There's no guarantee that the package content API's version list is sorted. (I'm not even sure this is intentional behavior). If you use this API, I would recommend parsing versions. Also, note that this API includes unlisted packages, which are likely not meant for consumption.

Alternatively, you could consider using NuGet's search API:

1. [Including prelease query](https://azuresearch-ussc.nuget.org/query?semVerLevel=2.0.0&prerelease=true&debug=true&q=packageid:Microsoft.Windows.SDK.Win32Metadata)
2. [Excluding prerelease query](https://azuresearch-ussc.nuget.org/query?semVerLevel=2.0.0&prerelease=true&debug=true&q=packageid:Microsoft.Windows.SDK.Win32Metadata)

You can find the latest version using `json['data'][0]['version']`. This will exclude unlisted packages. Search API docs: https://learn.microsoft.com/nuget/api/search-query-service-resource

By the way, NuGet uses service discovery and can move its resources at any time. This enables failover scenarios to maintain high availability globally, and, allows NuGet to make breaking changes to the protocol without affecting well-behaved clients. You can learn more about this here: https://learn.microsoft.com/nuget/api/overview

Ideally your app uses service discovery:
1. Get https://api.nuget.org/v3/index.json 
2. Grab the URL for the version of the resource it knows it is compatible with.
3. Uses that URL to do subsequent API requests

Here's an example of a well-conforming client: https://github.com/loic-sharma/BaGet/tree/main/src/BaGet.Protocol#search-for-packages

Here's where it discovers NuGet resources: https://github.com/loic-sharma/BaGet/blob/5a8502ca4e3b21acd369d4a98c2529dac2d81cfe/src/BaGet.Protocol/NuGetClientFactory.cs#L155-L163

Reply:
Thanks -- this is really helpful!